runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/http/JettyRequestAPIData.java

```diff
@@ -119,6 +119,12 @@ public JettyRequestAPIData(
 
     HttpFields.Mutable fields = HttpFields.build();
     for (HttpField field : request.getHeaders()) {
+      // If it has a HttpHeader it is one of the standard headers so won't match any appengine specific header.
+      if (field.getHeader() != null) {
+        fields.add(field);
+        continue;
+      }
```

Comment:
1. Actually even without this logic, a non-AE header should have been added by the [line below](https://github.com/GoogleCloudPlatform/appengine-java-standard/pull/308/files#diff-14f9fd007b6be5c9a0f036212935ee49c47a2e92c2cfd4b28a8b5240bdb76018L232) no? Am I missing something here since the `switch` is only to certain certain flags and not to prevent addition of fields?
2. If my above statement is wrong and that we still need to have this check, can we please this condition along with that line mentioned there? Just to have a more consistent flow of readability on what all fields are getting added?